### ISSUE TITLE
Implementation of MultiHeadAttention is incomplete.

### ISSUE THREAD

**Issue by saahiluppal** [tensorflow/addons#1940]
while MultiHeadAttention should have to pay attention to two types of attention masks,
1. That masks the padding (called padding mask)
2. That prevents positions from attending (called attention mask)

While current implementation seems to have only one mask i.e. that masks the padding (1st option).

**Comment by bhack** [comment]
Please if you want more feature help us to review https://github.com/tensorflow/community/pull/260.

**Comment by saberkun** [comment]
@saahiluppal Thank you! Yes, please help us review and discuss in the RFC.
The padding mask, causal_mask mask and segment mask can finally compose an attention mask tensor which is (batch size, ....., target sequence, source sequence). If we put mask composition logic outside the layer and pass the attention mask in, it should be flexible to cover use cases. Please correct me if I am wrong. (Yes, we can move the discussion to the RFC. Thanks!)

**Comment by seanpmorgan** [comment]
Closing as this discussion should continue in the new RFC. Thanks!
